File: in_java/Pattern_12.java
package in_java;

public class Pattern_12 extends Pattern {
    public Pattern_12(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        for (int i = 0; i < this.row; i++) {
            for (int j = 0; j < this.column; j++) {
                if (j >= i && j <= this.column - 1 - i) {
                    System.out.print(this.symbol);
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_12.rs
pub fn pattern_12() {
    let row = 5;
    let column = 9;
    let symbol = "*";

    for i in 0..row {
        for j in 0..column {
            if j >= i && j <= column - 1 - i {
                print!("{}", symbol);
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
